service/src/test/java/bio/terra/cbas/runsets/inputs/TestInputGeneratorBuildInputs.java

```diff
@@ -169,4 +173,56 @@ private static RecordResponse fooRatingRecord(String rawAttributeJson)
             .trim(),
         RecordResponse.class);
   }
+
+  @Test
+  void invalidAttribute() throws Exception {
+    String incorrectAttributeDefinition =
+        """
+            [
+              {
+                "input_name": "lookup_foo",
+                "input_type": { "type": "primitive", "primitive_type": "String" },
+                "source": {
+                  "type": "record_lookup",
+                  "record_attribute": "foo_rating"
+                }
+              }
+            ]
+          """
+            .stripIndent()
+            .trim();
+
+    RecordResponse recordResponse =
+        objectMapper.readValue(
+            """
+        {
+          "id": "FOO1",
+          "type": "FOO",
+          "attributes": {
+            "MY_RECORD_ATTRIBUTE": "Hello, world"
+          },
+          "metadata": {
+            "provenance": "TODO: RECORDMETADATA"
+          }
```

Comment:
Try rebasing again, because I think this is now going to need to be removed, due to WDS changing.

Reply:
does that make this a good candidate for contract tests in the future?

I'm not suggesting Katrina actually try to do it in this review cycle, just thinking ahead

Reply:
it absolutely does!